### PR TITLE
Added any motion interrupt support

### PR DESCRIPTION
**Pull request by Rocketct**
added support to any motion interrupt in MKRIMU library

cc/ @facchinm @sandeepmistry tested and works

### CODE REVIEW

File: src/IMU.cpp [hunk 1]
```diff
@@ -109,6 +114,58 @@ void IMUClass::end()
   _wire->end();
 }
 
+voidFuncPtr cb;
+IMUClass * imurq;
```
Comment (sandeepmistry) [inline src/IMU.cpp]: Having these as global vars is not good ...

File: src/IMU.h [hunk 2]
```diff
@@ -33,6 +33,11 @@ class IMUClass {
   int begin();
   void end();
 
+  //interrupt management
+  void attachInterrupt(voidFuncPtr callback); // Enable interrupt.
```
Comment (sandeepmistry) [inline src/IMU.h]: Is there only one type of interrupt the IMU supports?

Is there a way to configure the thresholds?

File: src/IMU.h [hunk 3]
```diff
@@ -33,6 +33,11 @@ class IMUClass {
...
+  void detachInterrupt();
+  static void HighMotionAcceleration();
```
Comment (sandeepmistry) [inline src/IMU.h]: Should this be public?

File: src/IMU.cpp [hunk 4]
```diff
@@ -109,6 +114,58 @@ void IMUClass::end()
...
+  imurq->writeRegister(BNNO055_SYS_TRIGGER_REG, 0x40);
+}
+
+void IMUClass::attachInterrupt(voidFuncPtr callback)
+{
+  cb = callback;
+  imurq = this;
+
+  // attach interrupt to pin 0
+  ::attachInterrupt(0, HighMotionAcceleration, RISING);
```
Comment (sandeepmistry) [inline src/IMU.cpp]: Should the pin be passed in the constructor?

File: src/IMU.cpp [hunk 5]
```diff
@@ -44,6 +44,11 @@
 #define BNNO055_SYS_TRIGGER_REG        0x3f
 #define BNNO055_AXIS_MAP_CONFIG_REG    0x41
 #define BNNO055_AXIS_MAP_SIGN_REG      0x42
+#define BNNO055_INT_EN_REG             0x10
+#define BNNO055_ACC_AM_THRES_REG       0x11
+#define BNNO055_ACC_INT_Settings_REG   0x12
```
Comment (sandeepmistry) [inline src/IMU.cpp]: consistent case?

File: src/IMU.cpp (hunk 4 above)
Comment (Rocketct) [reply src/IMU.cpp]: yes it will be, i'll add a variable and set as usual on the constructor

File: src/IMU.h (hunk 2 above)
Comment (Rocketct) [reply src/IMU.h]: yes there are other interrupt, yes there is a way to configure the threshold.
This was done for a demonstration in this way.

**Comment by aentinger** [comment]
I'm closing this due to inactivity. @Rocketct please re-open if you intend to complete this PR, otherwise let's leave it closed.

**Pull request closed without merging**